app/models/order.rb

```diff
@@ -51,6 +67,28 @@ def sufficient_funds_for_buy
     errors.add(:shares, "Insufficient funds. You have #{formatted_balance} but need #{formatted_cost}")
   end
 
+  def sufficient_funds_for_sell_when_update
+    # we would want to execute this method only when we are just updating the shares
+    # we are not marking the order as completed
+
```

Comment:
We don't want to allow any orders to be updated that aren't pending, so we should add a guard clause here that bails unless the status is pending so we bail out before doing any other logic.

Or we could add a separate validation which might be preferable something like:

validate :order_is_pending, on: :update

 ... 

def order_is_pending
  errors.add(:base, "Cannot update non-pending orders") unless pending?
end